docs/developer_guide/e2e_test.md

```diff
@@ -94,4 +94,36 @@ ginkgo run -r -v --skip-file "admission_controller_test.go" -- -yk-namespace "yu
 * Delete the cluster after we finish testing (this step is optional).
 ```shell
 ./scripts/run-e2e-tests.sh -a cleanup -n yktest
-```
\ No newline at end of file
+```
+
+## Checking E2E Test Logs
+
+After implementing [YUNIKORN-2305](https://issues.apache.org/jira/browse/YUNIKORN-2305), the logs of failed E2E tests are stored locally and uploaded to the GitHub Action Artifact. This section explains how to retrieve and understand these logs.
```

Comment:
We don't need to mention the JIRA that implemented this. It's fine to have it in the docs without that.

Reply:
Done.